### PR TITLE
Support accessing global memories in injecting LG templates as Expressions

### PR DESCRIPTION
**Pull request by cosmicshuai**
Issue reported by @chrimc62 that the injected LG function don't have access to global state (DSM state).

This PR enable injected LG template functions to access global memories. 

Suppose we have a template, with a global variable {user: {name: "user1", date: "2001-01-01"}, }
```
> !# @strict = false
> !# @namespace = Adaptive
> !# @Exports = UserDataConcat

#UserDataConcat()
- ${user.name} : ${user.date}
```

The result of evaluating the expression `Expression.parse("Adaptive.UserDataConcat()")` is `user1 : 2001-01-01`

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.LanguageGeneration.Tests/TemplatesTest.cs [hunk 1]
```diff
@@ -1481,12 +1481,21 @@ public void TestCustomFunction2()
         public void TestInjectLG()
         {
             var templates = Templates.ParseFile(GetExampleFilePath("./InjectionTest/inject.lg"));
-            
-            var (evaled, error) = Expression.Parse("foo.bar()").TryEvaluate(null);
-            
+
```
Comment (Danieladu) [inline tests/Microsoft.Bot.Builder.LanguageGeneration.Tests/TemplatesTest.cs:1484]: Maybe add more tests about the "Expression.TryEvaluate(IMemory state)" is better. 
for example:
```
var memory = StackedMemory.Wrap(...);
```
or 
```
var memory = new StackMemoory();
memory.push(new SimpleMemory(xxx))
memory.push(new CustomizedMeoory(xxx))
```

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs [hunk 2]
```diff
@@ -287,7 +287,7 @@ public object ConstructScope(string inputTemplateName, List<object> args)
             }
 
             var parameters = TemplateMap[templateName].Parameters;
-            var currentScope = _evaluationTargetStack.Count > 0 ? CurrentTarget().Scope : new CustomizedMemory(null);
+            var currentScope = CurrentTarget().Scope;
```
Comment (Danieladu) [inline libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs:290]: Why change this one.

Comment (cosmicshuai) [reply libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs:290]: This is the original code. In last PR for https://github.com/microsoft/botbuilder-dotnet/pull/3802, I changed this line to make sure constructScope can work in injecting LG, where _evaluationTargetStack.Count is 0.
In current solution, we don't need constructScope  any more. So I modified the code to the original.

File: tests/Microsoft.Bot.Builder.LanguageGeneration.Tests/TemplatesTest.cs (hunk 1 above)
Comment (cosmicshuai) [reply tests/Microsoft.Bot.Builder.LanguageGeneration.Tests/TemplatesTest.cs:1484]: tests added

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Templates.Tests/lg/inject.lg [hunk 3]
```diff
@@ -13,4 +13,7 @@
 -user.tasks
 
 #GetMessage
-- This is an injected message
\ No newline at end of file
+- This is an injected message
+
+#UserDataConcat()
```
Comment (boydc2014) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Templates.Tests/lg/inject.lg:18]: can you make this test case or create new test case to be able to test both local and global case?

**Pull request merged**